lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -202,13 +203,16 @@ class StableHLOToTTIRReshapeOpConversionPattern
       new_shape_i32.push_back(static_cast<int32_t>(dim));
     }
     ArrayAttr new_shape_attr = rewriter.getI32ArrayAttr(new_shape_i32);
-    rewriter.replaceOpWithNewOp<mlir::tt::ttir::ReshapeOp>(
+    auto new_reshape_op = rewriter.replaceOpWithNewOp<mlir::tt::ttir::ReshapeOp>(
         srcOp, getTypeConverter()->convertType(outputTensor.getType()),
-        srcOp->getOperand(0), outputTensor, new_shape_attr,
+        adaptor.getOperand(), outputTensor, new_shape_attr,
         rewriter.getArrayAttr(
             SmallVector<Attribute>(adaptor.getOperands().size() + 1,
                                    rewriter.getAttr<OperandConstraintAttr>(
                                        OperandConstraint::AnyDeviceTile))));
+    if (new_reshape_op.getType() == new_reshape_op->getOperand(0).getType()) {
```

Comment:
A comment would be nice to explain this non-obvious statement.

Reply:
Fixed in the next commit.